script.js

```diff
@@ -0,0 +1,959 @@
+
+  // Define the column headers for rows
+  const rowElement = document.createElement('tr');
+  tableScoreCard.appendChild(rowElement);
+
+  const rowHeaderGameCount = document.createElement('th');
+  rowHeaderGameCount.innerText = 'Game Counter';
+  rowElement.appendChild(rowHeaderGameCount);
+
+  const rowHeaderGameResult = document.createElement('th');
+  rowHeaderGameResult.innerText = 'Result';
+  rowElement.appendChild(rowHeaderGameResult);
+
+  const rowHeaderBetAmount = document.createElement('th');
+  rowHeaderBetAmount.innerText = 'Bet Amount';
+  rowElement.appendChild(rowHeaderBetAmount);
+
+  const rowHeaderCreditsPlayed = document.createElement('th');
+  rowHeaderCreditsPlayed.innerText = 'Total Credits Played';
+  rowElement.appendChild(rowHeaderCreditsPlayed);
+
+  // const rowHeaderCreditsWon = document.createElement('th');
+  // rowHeaderCreditsWon.innerText = 'Credits Won';
+  // rowElement.appendChild(rowHeaderCreditsWon);
+
+  const rowHeaderTotalCreditsWon = document.createElement('th');
+  rowHeaderTotalCreditsWon.innerText = 'Total Credits Won';
+  rowElement.appendChild(rowHeaderTotalCreditsWon);
+
+  const rowHeaderWinLoss = document.createElement('th');
+  rowHeaderWinLoss.innerText = 'Win-Loss Count';
+  rowElement.appendChild(rowHeaderWinLoss);
+
+  const rowHeaderFinalCardSet = document.createElement('th');
+  rowHeaderFinalCardSet.innerText = 'Final Set of Cards Played';
+  rowElement.appendChild(rowHeaderFinalCardSet);
+};
+
+// Function to create all the elements for displaying the Score Card
+const createResultContainerElements = () => {
+  // Main container to hold the result of the game
+  const divGameResultContainer = document.createElement('div');
+  divGameResultContainer.classList.add('result');
+
+  // Button to show and hide the result details
+  const divScoreButtonContainer = document.createElement('div');
+  buttonShowHideScore = document.createElement('button');
+  buttonShowHideScore.classList.add('button-common', 'stand-alone-button');
+  buttonShowHideScore.innerText = `${TXT_SHOW_RESULT}`;
+  divScoreButtonContainer.appendChild(buttonShowHideScore);
+  divGameResultContainer.appendChild(divScoreButtonContainer);
+
+  // Container to hold the result
+  const divScoreContainer = document.createElement('div');
+  divScoreContainer.classList.add('score-card-container');
+  // In the intial Stage this container will not be displayed.
+  divScoreContainer.style.display = 'none';
+  divGameResultContainer.appendChild(divScoreContainer);
+
+  // Associate the button to the container display function
+  buttonShowHideScore.addEventListener('click', () => {
+    toggleElementDisplay(divScoreContainer);
+  });
+  // Table that shows the whole result of the game
+  createScoreCardTable();
+  divScoreContainer.appendChild(tableScoreCard);
+  document.body.appendChild(divGameResultContainer);
+};
+
+// Function that initializes the DOM element during the game load
+const gameInit = () => {
```

Comment:
Great work decomposing UI creation into helper functions and separating UI from game logic into different functions wherever possible!